job/common/src/main/java/alluxio/job/wire/PlanInfo.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.job.wire;
+
+import alluxio.exception.status.InvalidArgumentException;
+import alluxio.grpc.JobType;
+import alluxio.job.util.SerializationUtils;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.protobuf.ByteString;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * The job descriptor.
+ */
+public final class PlanInfo implements JobInfo {
+  private final long mId;
+  private final String mName;
+  private final String mErrorMessage;
+  private final List<JobInfo> mChildren;
+  private final Status mStatus;
+  private final String mResult;
+  private final long mLastUpdated;
+
+  /**
+   * JobInfo constructor exposed for testing.
+   * @param id job id
+   * @param name name of the job
+   * @param status job status
+   * @param lastUpdated last updated time in milliseconds
+   * @param errorMessage job error message
+   */
+  public PlanInfo(long id, String name, Status status, long lastUpdated,
+                  @Nullable String errorMessage) {
+    mId = id;
+    mName = name;
+    mStatus = status;
+    mLastUpdated = lastUpdated;
+    mErrorMessage = (errorMessage == null) ? "" : errorMessage;
+    mChildren = ImmutableList.of();
+    mResult = null;
+  }
+
+  /**
+   * Constructs the job info from the job master's internal representation of job info.
+   *
+   * @param jobInfo the job master's internal job info
```

Comment:
Yes, there is quite a bit more renaming that needs to be done with the internal structures.